src/MongoDbEventStoreAdapter.php

```diff
@@ -173,9 +173,15 @@ public function appendTo(StreamName $streamName, Iterator $streamEvents)
 
         try {
             $insertBatch->execute();
-        } catch (\MongoWriteConcernException $e) {
+        } catch (\MongoDuplicateKeyException $e) {
             throw new ConcurrencyException('', 0, $e);
```

Comment:
We should add a short exception message so that one knows what went wrong when the exception is logged.